src/main/java/de/tum/in/www1/exerciseapp/service/QuizExerciseService.java

```diff
@@ -191,10 +192,31 @@ public void adjustResultsOnQuizDeletions(QuizExercise quizExercise) {
                         // Check if an answerOption was deleted and delete reference to in selectedOptions
                         ((MultipleChoiceSubmittedAnswer) submittedAnswer).checkForDeletedAnswerOptions((MultipleChoiceQuestion) question);
                     }
-                    // TODO: @Moritz: DragAndDrop Question
+                }
+                if (submittedAnswer instanceof DragAndDropSubmittedAnswer) {
+                    // Delete all references to question, dropLocations and dragItem if the question was deleted
+                    if (!quizExercise.getQuestions().contains(submittedAnswer.getQuestion())) {
+                        submittedAnswer.setQuestion(null);
+                        ((DragAndDropSubmittedAnswer) submittedAnswer).setMappings(null);
```

Comment:
Please do not cast several times, directly cast ones after the if statement and save the casted object in a new variable